adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/lwm2m/LwM2MResourceDirectory.java

```diff
@@ -0,0 +1,694 @@
+                ctx.getTenantId(), ctx.getOriginDevice().getDeviceId());
+        currentSpan.log(Map.of(
+                "lwm2m", lwm2mVersion,
+                "endpoint", endpoint,
+                "binding mode", bindingMode.name(),
+                "lifetime (secs)", lifetime));
+
+        if (Strings.isNullOrEmpty(lwm2mVersion)) {
+            result.fail(new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST,
+                    QUERY_PARAM_LWM2M_VERSION + " query parameter must be non-empty"));
+        } else if (!lwm2mVersion.startsWith("1.0")) {
+            result.fail(new ClientErrorException(HttpURLConnection.HTTP_PRECON_FAILED,
+                    "server supports LwM2M version 1.0.x only"));
+        } else if (Strings.isNullOrEmpty(endpoint)) {
+            result.fail(new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST,
+                    QUERY_PARAM_ENDPOINT + " query parameter must be non-empty"));
+        } else if (lifetime <= 0) {
+            result.fail(new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST,
+                    QUERY_PARAM_LIFETIME + " query parameter must be positive integer"));
+        } else if (!supportedBindingModes.contains(bindingMode)) {
+            result.fail(new ClientErrorException(HttpURLConnection.HTTP_PRECON_FAILED,
+                    "server supports binding modes U and UQ only"));
+        } else {
+            final var identity = EndpointContextUtil.extractIdentity(ctx.getExchange().advanced().getRequest().getSourceContext());
+            final var registration = new Registration.Builder(getRegistrationId(ctx.getOriginDevice()), endpoint, identity)
+                    .lwM2mVersion(lwm2mVersion)
+                    .lifeTimeInSec(lifetime)
+                    .bindingMode(bindingMode)
+                    .objectLinks(objectLinks)
+                    .additionalRegistrationAttributes(Map.of(
+                            TracingHelper.TAG_TENANT_ID.getKey(), ctx.getTenantId(),
+                            TracingHelper.TAG_DEVICE_ID.getKey(), ctx.getAuthenticatedDevice().getDeviceId()))
+                    .build();
+            LOG.debug("adding {}", registration);
+            lwm2mRegistrationStore.addRegistration(registration);
+
+            getAdapter().getCommandConsumerFactory()
+                .createCommandConsumer(
+                    ctx.getTenantId(),
+                    ctx.getOriginDevice().getDeviceId(),
+                    commandContext -> {
+                        Tags.COMPONENT.set(commandContext.getTracingSpan(), getAdapter().getTypeName());
+                     // TODO add logic for sending commands to device
+                    },
+                    null,
+                    currentSpan.context())
+                .compose(commandConsumer -> {
+                    commandConsumers.put(registration.getId(), commandConsumer);
+                    return sendLifetimeAsTtdEvent(ctx, registration, currentSpan);
```

Comment:
The "lwm2m lifetime" is in my opinion no "hono-ttd". that time defines, how frequently update-registrations are expected, not how long the device stays receivable